pkg/route/validation/validation_test.go

```diff
@@ -1200,6 +1854,120 @@ func TestValidateRouteUpdate(t *testing.T) {
+		{description: "parameters for a sample fetcher that doesn't take parameters", input: `%[ssl_c_der(host)]`, validInput: false},
+		{description: "dangerous sample fetchers and converters", input: `%[env(FOO)]`, validInput: false},
+		{description: "dangerous sample fetchers and converters", input: `%[req.hdr(host),debug()]`, validInput: false},
+		{description: "extra comma", input: `%[req.hdr(host),,lower]`, validInput: false},
+
+		// CR and LF are not allowed in header value as per RFC https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.4
+		{description: "carriage return", input: "\r", validInput: false},
+		{description: "CRLF", input: "\r\n", validInput: false},
+
+		{description: "environment variable with a bracket missing", input: `${NET_COOLOCP_HOSTPRIMARY`, validInput: true},
+		{description: "value with conditional expression and env var", input: `%[XYZ.hdr(Host)] if ${NET_COOLOCP_HOSTPRIMARY`, validInput: true},
+		{description: "value with what looks like a conditional expression and env var", input: `%[XYZ.hdr(Host)]\ if\ ${NET_COOLOCP_HOSTPRIMARY`, validInput: true},
+
+		{description: "sample value", input: `%ci:%cp [%tr] %ft %ac/%fc %[fc_err]/%[ssl_fc_err,hex]/%[ssl_c_err]/%[ssl_c_ca_err]/%[ssl_fc_is_resumed] %[ssl_fc_sni]/%sslv/%sslc`, validInput: false},
+		{description: "interpolation of T i.e %T", input: `%T`, validInput: false},
+
+		// url
+		// regex does not check validity of url in a header value.
+		{description: "hdr with url", input: `http:??//url/hack`, validInput: true},
+
+		// spaces and tab
+		// regex allows spaces before and after. The reason is that after a dynamic value is provided someone might provide a condition `%[XYZ.hdr(Host)] if foo` which would have
+		// spaces after the dynamic value and if condition.
+		// tab is rejected as control characters are not allowed by the regex.
+		{description: "space before and after the value", input: ` T `, validInput: true},
+		{description: "double space before and after the value", input: `  T  `, validInput: true},
+		{description: "tab before and after the value", input: `	T	`, validInput: false},
+		{description: "tab before and after the value", input: "\tT\t", validInput: false},
+	}
+
+	var requestTypes = []struct {
+		description          string
+		regexp               *regexp.Regexp
+		testInputSubstituter func(s string) string
+	}{{
+		description:          "request",
+		regexp:               permittedRequestHeaderValueRE,
+		testInputSubstituter: func(s string) string { return strings.ReplaceAll(s, "XYZ", "req") },
+	}, {
+		description:          "response",
+		regexp:               permittedResponseHeaderValueRE,
+		testInputSubstituter: func(s string) string { return strings.ReplaceAll(s, "XYZ", "res") },
+	}}
+
+	for _, rt := range requestTypes {
+		t.Run(rt.description, func(t *testing.T) {
+			for _, tc := range tests {
+				t.Run(tc.description, func(t *testing.T) {
+					input := rt.testInputSubstituter(tc.input)
```

Comment:
Shouldn't this be calling validateHeaders?

Reply:
no this test was written just test various types of string inputs using the regex we use. So we created a function and put out regex we use to test the input values.

Reply:
done